### PR TITLE
Add new fields and Snapshot model for localunit

### PR DESCRIPTION
**Pull request by susilnem**
Addresses 
- https://github.com/IFRCGo/go-api/issues/2328
- https://github.com/IFRCGo/go-api/issues/2329

## Changes
- Add fields for local units
- Add new model for snapshot of localunit

## Checklist
Things that should succeed before merging.

- [ ] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

**Review by Rup-Narayan-Rajbanshi — APPROVED** [review]

File: local_units/models.py [hunk 1]
```diff
@@ -277,6 +277,16 @@ def __str__(self):
 
 @reversion.register(follow=("health",))
 class LocalUnit(models.Model):
+    class Status(models.IntegerChoices):
+        VERIFIED = 1, _("Verified")
+        UNVERIFIED = 2, _("Unverified")
+
+    class DepricateReason(models.IntegerChoices):
```
Comment (Rup-Narayan-Rajbanshi) [inline local_units/models.py:284]:
```suggestion
    class DeprecateReason(models.IntegerChoices):
```

File: local_units/models.py [hunk 2]
```diff
@@ -340,12 +350,70 @@ class LocalUnit(models.Model):
     email = models.EmailField(max_length=255, blank=True, null=True, verbose_name=_("Email"))
     link = models.URLField(max_length=255, blank=True, null=True, verbose_name=_("Social link"))
     location = models.PointField(srid=4326, help_text="Local Unit Location")
+    status = models.IntegerField(
+        choices=Status.choices, verbose_name=_("status"), default=Status.UNVERIFIED, blank=True, null=True
+    )
+    is_deprecated = models.BooleanField(default=False, verbose_name=_("Is deprecated?"), blank=True, null=True)
+    deprecated_reason = models.IntegerField(
+        choices=DepricateReason.choices, verbose_name=_("deprecated reason"), blank=True, null=True
```
Comment (Rup-Narayan-Rajbanshi) [inline local_units/models.py:358]:
```suggestion
        choices=DeprecateReason.choices, verbose_name=_("deprecated reason"), blank=True, null=True
```

**Review by Rup-Narayan-Rajbanshi — COMMENTED** [review]
lets create the migration file again.

File: local_units/models.py [hunk 3]
```diff
@@ -340,12 +350,70 @@ class LocalUnit(models.Model):
     email = models.EmailField(max_length=255, blank=True, null=True, verbose_name=_("Email"))
     link = models.URLField(max_length=255, blank=True, null=True, verbose_name=_("Social link"))
     location = models.PointField(srid=4326, help_text="Local Unit Location")
+    status = models.IntegerField(
+        choices=Status.choices, verbose_name=_("status"), default=Status.UNVERIFIED, blank=True, null=True
```
Comment (thenav56) [inline local_units/models.py:354]: Do we need black/null?

File: local_units/models.py [hunk 4]
```diff
@@ -340,12 +350,70 @@ class LocalUnit(models.Model):
...
+    )
+    is_deprecated = models.BooleanField(default=False, verbose_name=_("Is deprecated?"), blank=True, null=True)
```
Comment (thenav56) [inline local_units/models.py:356]: Do we need black/null?

File: local_units/models.py [hunk 5]
```diff
@@ -340,12 +350,70 @@ class LocalUnit(models.Model):
...
+
+    # NOTE: triggered_by is the user who created the request
+    triggered_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        verbose_name=_("triggered by"),
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="tiggered_by_local_unit",
+    )
+    triggered_at = models.DateTimeField(verbose_name=_("Triggered at"), auto_now=True)
```
Comment (thenav56) [inline local_units/models.py:398]: Use auto_now_add=
https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.DateField.auto_now_add

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**